crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -460,28 +539,63 @@ impl FromWorld for MeshPipeline {
         };
 
         let mesh_layout = render_device.create_bind_group_layout(&BindGroupLayoutDescriptor {
-            entries: &[mesh_binding],
+            entries: &[mesh_binding(0)],
             label: Some("mesh_layout"),
         });
-
+        let skinned_mesh_binding = |binding| BindGroupLayoutEntry {
+            binding,
+            visibility: ShaderStages::VERTEX,
+            ty: BindingType::Buffer {
+                ty: BufferBindingType::Uniform,
+                has_dynamic_offset: true,
+                min_binding_size: BufferSize::new(JOINT_BUFFER_SIZE as u64),
+            },
+            count: None,
+        };
         let skinned_mesh_layout =
+            render_device.create_bind_group_layout(&BindGroupLayoutDescriptor {
+                entries: &[mesh_binding(0), skinned_mesh_binding(1)],
+                label: Some("skinned_mesh_layout"),
+            });
+        let morph_targets_weights_binding = |binding| BindGroupLayoutEntry {
+            binding,
+            visibility: ShaderStages::VERTEX,
+            ty: BindingType::Buffer {
+                ty: BufferBindingType::Uniform,
+                has_dynamic_offset: true,
+                min_binding_size: BufferSize::new(MORPH_BUFFER_SIZE as u64),
+            },
+            count: None,
+        };
+        let morph_targets_diff_binding = |binding| BindGroupLayoutEntry {
+            binding,
+            visibility: ShaderStages::VERTEX,
+            ty: BindingType::StorageTexture {
+                access: StorageTextureAccess::ReadOnly,
+                format: TextureFormat::R32Float,
+                view_dimension: TextureViewDimension::D3,
+            },
+            count: None,
+        };
+        let morph_targets_layout =
             render_device.create_bind_group_layout(&BindGroupLayoutDescriptor {
                 entries: &[
-                    mesh_binding,
-                    BindGroupLayoutEntry {
-                        binding: 1,
-                        visibility: ShaderStages::VERTEX,
-                        ty: BindingType::Buffer {
-                            ty: BufferBindingType::Uniform,
-                            has_dynamic_offset: true,
-                            min_binding_size: BufferSize::new(JOINT_BUFFER_SIZE as u64),
-                        },
-                        count: None,
-                    },
+                    mesh_binding(0),
+                    morph_targets_weights_binding(2),
```

Comment:
I'm not sure if you're allowed to skip binding indices like this.

Reply:
I just tested it, and it seems to work fine with gaps. I prefer keeping the gap, because otherwise I have to do a weird dance with `#ifdef`s in the shader code.